x/posts/internal/types/post_test.go

```diff
@@ -760,6 +760,93 @@ func TestPost_Equals(t *testing.T) {
 	}
 }
 
+func TestPost_Unique(t *testing.T) {
+
+	tests := []struct {
+		name       string
+		input      []string
+		expectedIn []string
+	}{
+		{
+			name:       "Unique returns unique strings slice",
+			input:      []string{"hello", "hello", "world", "world"},
+			expectedIn: []string{"hello", "world"},
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			actualIn := types.Unique(test.input)
+			require.Equal(t, test.expectedIn, actualIn)
+		})
+	}
+}
+
+func TestPost_GetPostHashtags(t *testing.T) {
+	owner, err := sdk.AccAddressFromBech32("cosmos1cjf97gpzwmaf30pzvaargfgr884mpp5ak8f7ns")
+	require.NoError(t, err)
+
+	timeZone, err := time.LoadLocation("UTC")
+	require.NoError(t, err)
+
+	date := time.Date(2020, 1, 1, 12, 00, 00, 000, timeZone)
+
+	tests := []struct {
+		name        string
+		post        types.Post
+		expHashtags []string
+	}{
+		{
+			name: "Hashtags in message extracted correctly",
+			post: types.NewPost(types.PostID(1),
+				types.PostID(0),
+				"Post with #test #desmos",
+				false,
+				"4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e",
+				map[string]string{},
+				date,
+				owner,
+			),
+			expHashtags: []string{"#test", "#desmos"},
+		},
+		{
+			name: "No hashtags in message",
+			post: types.NewPost(types.PostID(1),
+				types.PostID(0),
+				"Post with no hashtag",
+				false,
+				"4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e",
+				map[string]string{},
+				date,
+				owner,
+			),
+			expHashtags: []string{},
+		},
+		{
+			name: "No same hashtags inside string array",
+			post: types.NewPost(types.PostID(1),
+				types.PostID(0),
+				"Post with double #hashtag #hashtag",
+				false,
+				"4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e",
+				map[string]string{},
+				date,
+				owner,
+			),
+			expHashtags: []string{"#hashtag"},
+		},
```

Comment:
I think it's better to make so that `GetPostHashtags` returns the hashtags **without** the leading `#`.

```suggestion
		{
			name: "Hashtags in message extracted correctly",
			post: types.NewPost(types.PostID(1),
				types.PostID(0),
				"Post with #test #desmos",
				false,
				"4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e",
				map[string]string{},
				date,
				owner,
			),
			expHashtags: []string{"test", "desmos"},
		},
		{
			name: "No hashtags in message",
			post: types.NewPost(types.PostID(1),
				types.PostID(0),
				"Post with no hashtag",
				false,
				"4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e",
				map[string]string{},
				date,
				owner,
			),
			expHashtags: []string{},
		},
		{
			name: "No same hashtags inside string array",
			post: types.NewPost(types.PostID(1),
				types.PostID(0),
				"Post with double #hashtag #hashtag",
				false,
				"4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e",
				map[string]string{},
				date,
				owner,
			),
			expHashtags: []string{"hashtag"},
		},
```